src/nvim/option.c

```diff
@@ -2042,10 +2042,9 @@ static char *set_bool_option(const int opt_idx, char_u *const varp, const int va
     }
     redraw_titles();
     modified_was_set = value;
-  }
 
 #ifdef BACKSLASH_IN_FILENAME
-  else if ((int *)varp == &p_ssl) {
+  } else if ((int *)varp == &p_ssl) {
```

Comment:
This is changing the scope. To make sure this does not change the semantics, we need to check all permutations if the `#ifdef` layers. Even if you did that, it could cause a subtle bug when merging a vim patch later. That's why I don't think this change is worth the risk.

Reply:
It may be easier to understand if you remove the blank line above.

Reply:
> The closing } brace was removed from the previous area.

Yes and moved to line 2063.

I agree some of these ifdef changes are a little risky, and I've backed off on some, but I think this one is ok.